src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSourceUnitTest.java

```diff
@@ -187,6 +186,21 @@ public void testCRAMReferenceFromHDFS() throws Exception {
         });
     }
 
+    @Test
+    public void testIntervals(){
+        JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
+        ReadsSparkSource readSource = new ReadsSparkSource(ctx);
+        List<SimpleInterval> intervals =
+                ImmutableList.of(new SimpleInterval("17", 69010, 69040), new SimpleInterval("17", 69910, 69920));
+        JavaRDD<GATKRead> reads = readSource.getParallelReads(NA12878_chr17_1k_BAM, null, intervals);
+
+        SamReaderFactory samReaderFactory = SamReaderFactory.makeDefault().validationStringency(ValidationStringency.SILENT);
+        SamReader samReader = samReaderFactory.open(new File(NA12878_chr17_1k_BAM));
+        int seqIndex = samReader.getFileHeader().getSequenceIndex("17");
+        SAMRecordIterator query = samReader.query(new QueryInterval[]{new QueryInterval(seqIndex, 69010, 69040), new QueryInterval(seqIndex, 69910, 69920)}, false);
+        Assert.assertEquals(reads.count(), Iterators.size(query));
+    }
+
```

Comment:
It's probably worth adding a test case with the large interval list I linked to above to make sure there are no performance problems there (should probably be checked in to `src/test/resources/large` using `git lfs`). Can use the methods in `IntervalArgumentCollection` to load the interval list.